server/pbench/scripts/unpack-tarballs

```diff
@@ -0,0 +1,91 @@
+#! /bin/bash
+
+# we asssume that these scripts are installed in /opt/pbench-server/scripts
+# on the server.
+
+###########################################################################
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))/../bin
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)/../bin
+        PROG=$(basename $0)
+        ;;
+esac
+
+# TOP, ARCHIVE, INCOMING, RESULTS, USERS are all defined by the base file
+. $dir/pbench-base.sh
+###########################################################################
+ 
+controller=$1
+if [ -z "$controller" ] ;then
+   exit 1
+fi
+shift
+tarballs="$@"
+
+cd $ARCHIVE/$controller || exit 1
+
+incoming=$INCOMING/$controller
+
+linksrc=MOVE-UNPACKED
+linkdest=TO-UNPACK
+
+tblist=/tmp/unpack-tarballs.$$
+
+trap "rm -f $tblist" EXIT QUIT INT
+
+> $tblist
+if [ -z "$tarballs" ] ;then
+    ls *.tar.xz > $tblist
+else
+    for tb in $tarballs ;do
+        echo $tb >> $tblist
+    done
+fi
+
+while read tb ;do
+    # tb is a simple name: foo.tar.xz, not a path.
+    resultname=${tb%.tar.xz}
+
+    if [ ! -f "$resultname.tar.xz.md5" ] ;then
```

Comment:
Does this assume that the tar balls given the command line are in the archive controller area and are given as the full path there?

Reply:
This needs more work: I'm assuming a whole lot that is not necessarily true. Update coming.